bitcoin/src/blockdata/transaction.rs

```diff
@@ -773,14 +774,28 @@ impl Decodable for Transaction {
 ///
 /// * `fee_rate` - the fee rate of the transaction being created.
 /// * `input_weight_prediction` - the predicted input weight.
+///
+/// # Returns
+///
+/// This will return [`NumOpResult::Error`] if the fee calculation (fee_rate * weight) overflows.
+/// Otherwise, [`NumOpResult::Valid`] will wrap the successful calculation.
 pub fn effective_value(
     fee_rate: FeeRate,
     input_weight_prediction: InputWeightPrediction,
     value: Amount,
-) -> Option<SignedAmount> {
+) -> NumOpResult<SignedAmount> {
     let weight = input_weight_prediction.total_weight();
-    let signed_input_fee = fee_rate.to_fee(weight).ok()?.to_signed();
-    value.to_signed().checked_sub(signed_input_fee)
+
+    let fee = match fee_rate.to_fee(weight) {
+        NumOpResult::Valid(x) => x.to_signed(),
+        NumOpResult::Error(e) => return NumOpResult::Error(e)
+    };
+
+    // it's not possible to overflow when subtracted by the smallest given:
+    // Amount::MIN - SignedAmount::MAX = SignedAmount::MIN
+    // 
+    // Currently there is no API mechanisms to bypass the check in an efficient manner.
```

Comment:
nit: This is way too wordy for me (and I'm way to wordy in general), I'd prefer to just see

```rust
    value.to_signed() - fee     // Cannot overflow.
```

Reply:
Sure, I updated it.  Part of me worries I'll forget why it doesn't overflow but it's simple enough to figure out again.